sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -93,17 +97,19 @@ def upload_block_blob(  # pylint: disable=too-many-locals, too-many-statements
         # Do single put if the size is smaller than or equal config.max_single_put_size
         if adjusted_count is not None and (adjusted_count <= blob_settings.max_single_put_size):
             try:
-                data = data.read(length)
+                data = data.read(length)  #type: ignore
```

Comment:
This whole block of code needs to be reworked but I am unsure how to best do it. The issue is that it currently is "try and if it breaks then we know it wasn't that":
azure\storage\blob\_upload_helpers.py:99: error: Item "bytes" of "Union[bytes, str, Iterable[str], IO[str]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Item "str" of "Union[bytes, str, Iterable[str], IO[str]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Item "Iterable[str]" of "Union[bytes, str, Iterable[str], IO[str]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Item "bytes" of "Union[bytes, str, Iterable[bytes], IO[bytes]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Item "str" of "Union[bytes, str, Iterable[bytes], IO[bytes]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Item "Iterable[bytes]" of "Union[bytes, str, Iterable[bytes], IO[bytes]]" has no attribute "read"  [union-attr]
azure\storage\blob\_upload_helpers.py:99: error: Argument 1 to "read" of "IO" has incompatible type "Optional[int]"; expected "int"  [arg-type]